plugins/modules/ipaautomountlocation.py

```diff
@@ -0,0 +1,133 @@
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.0",
+    "supported_by": "community",
+    "status": ["preview"],
+}
+
+
+DOCUMENTATION = '''
+---
+module: ipa_automountlocation
+author: chris procter
+short_description: Manage FreeIPA autommount locations
+description:
+- Add and delete an IPA automount location
+options:
+  ipaadmin_principal:
+    description: The admin principal
+    default: admin
+  ipaadmin_password:
+    description: The admin password
+    required: false
+  name:
+    description: The automount location to be managed
+    required: true
+    aliases: ["cn","location"]
+  state:
+    description: State to ensure
+    required: false
+    default: present
+    choices: ["present", "absent"]
+'''
+
+EXAMPLES = '''
+  - name: ensure a automount location named DMZ exists
+    ipaautomountlocation:
+      ipaadmin_password: password01
+      name: DMZ
+      state: present
+
+  - name: remove an automount location named DMZ if it exists
+    ipaautomountlocation:
+      ipaadmin_password: password01
+      name: DMZ
+      state: absent
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.ansible_freeipa_module import FreeIPABaseModule
+
+
+class AutomountLocation(FreeIPABaseModule):
+
+    ipa_param_mapping = {}
+
+    def get_location(self, location):
+        response = self.api_command("automountlocation_find",
+                                    location,
+                                    args={"cn": location})
+
+        if response["count"] == 1:
+            location = response["result"][0]
+        else:
+            location = None
+        return location
+
+    def define_ipa_commands(self):
+        location = self.get_location(self.ipa_params.name)
+
+        if not location and self.ipa_params.state == "present":
+            # does not exist and is wanted
+            self.add_ipa_command(
+                "automountlocation_add",
+                name=self.ipa_params.name,
+                args=None,
+            )
+        elif location and self.ipa_params.state == "absent":
+            # exists and is not wanted
+            self.add_ipa_command(
+                "automountlocation_del",
+                name=self.ipa_params.name,
+                args=None,
+            )
+
+
+def main():
+    ipa_module = AutomountLocation(
+        argument_spec=dict(
+            ipaadmin_principal=dict(type="str",
+                                    default="admin"
+                                    ),
+            ipaadmin_password=dict(type="str",
+                                   required=False,
+                                   no_log=True
+                                   ),
+            state=dict(type='str',
+                       default='present',
+                       choices=['present', 'absent']
+                       ),
+            name=dict(type="str",
```

Comment:
As we support list of names when ensuring objects are absent, the type for `name` must be `list`.